UnicornRender/video/source/Material.cpp

```diff
@@ -104,5 +115,35 @@ glm::vec3 Material::GetColor() const
     return m_color;
 }
 
+void Material::SetTextureCoordinates(int32_t x, int32_t y, int32_t width, int32_t height)
+{
+    m_spriteCoords = { x, y, width, height };
+
+    NormalizeTextureCoordinates();
+
+    DataUpdated.emit();
+}
+
+glm::vec4 Material::GetTextureCoordinates() const
+{
+    return m_spriteCoords;
+}
+
+glm::vec4 Material::GetNormalizedTextureCoordinates() const
+{
+    return m_normSpriteCoords;
+}
+
+void Material::NormalizeTextureCoordinates()
+{
+    if (m_albedo->IsLoaded())
+    {
+        m_normSpriteCoords = { static_cast<float>(m_spriteCoords.x) / m_albedo->Width(),
```

Comment:
`m_spriteCoords` is a `glm::vec4`, why are you casting it to `float`? I feel like it makes more sense to cast `m_albedo->Width()` to `float`